src/puya/resolve-puya-path.ts

```diff
@@ -1,10 +1,14 @@
 import { Constants } from '../constants'
 import { logger } from '../logger'
-import { downloadPuyaBinary, findCachedPuyaBinary } from './puya-binary'
+import { createLockFile } from '../util/lock-file'
+import { downloadPuyaBinary, findCachedPuyaBinary, getPuyaStorageDir } from './puya-binary'
 import { parseSemVer } from './semver'
 
 export async function resolvePuyaPath(): Promise<string> {
   const version = parseSemVer(Constants.targetedPuyaVersion)
+  const puyaStorageDir = getPuyaStorageDir(version)
+
+  await using _ = await createLockFile(`${puyaStorageDir}.lock`, { maxRetries: 30, delayMs: 1000, staleMs: 5 * 60 * 1000 })
```

Comment:
5 minutes is probably a little too long for the staleness value. 1 minute is probably enough.